### ISSUE TITLE
Редирект сразу на страницу оплаты платежки

### ISSUE THREAD

**Issue by ElonaA** [wayforpay/Word-Press-Woocommerce#25]
Здравствуйте. Проблема актуальна на всех сайтах Вордпресс+Вукоммерс. Как можно убрать страницу типа /checkout/order-pay/7727/?key=wc_order_565f68f04d9ab&order=5253 на которую направляет сразу после оформления заказа, а потом только на платежку? После оформления сразу нужно кидать на страницу оплаты шлюза.

**Comment by ElonaA** [comment]
> Вибачаюсь, якщо лізу не в свою справу. А може це вам допоможе? Протестував в мене (після натискання кнопки "Підтвердити замовлення": Сторінка 1) https://secure.wayforpay.com/page?vkh=6347dfac-f7b8-4189-9379-080422d1dba8 На цій сторінці я вибрав картку.
> 
> Сторінка 2) https://secure.wayforpay.com/card/form?vkh=6347dfac-f7b8-4189-9379-080422d1dba8 На цій сторінці очикую підтвердження платежу.
> 
> [Скріншоти тут](https://imgur.com/a/wVo0XoS) (сайт надійний)
> 
> З повагою, Wlodz Mur

Тобто у вас одразу з чекаута після кнопки підтвердити перекидує на сторінку оплати? Чи трохи не зрозуміла.

**Comment by iWpress** [comment]
Так, я виклав два скріншота, першій це куди я попадаю після натискання на кнопку підтвердження.

**Comment by iWpress** [comment]
Чесно кажучи нічого особливого не робив. Встановив та використовую тестові ключи. Я трохи вносив зміни в плагін, але це тількі відображення на сайті і тільки зараз я виклав реквест.
Може все залежить від шаблона? 
Ще monobank виклав свій плагін на: [wordpress.org](https://wordpress.org/plugins/monopay/). Спробуйте його.

**Comment by ElonaA** [comment]
Не знаю, в мене стандартна Астра безкоштовна. Але на інших сайтах інші теми, та результат такий самий. Монобанк не підійде, бо треба вейфопей сам, тому дякую

**Comment by iWpress** [comment]
Спробуйте тут: iwpress.co.ua. Сайт тестовий (ключі тестові, але може списати гроші, тому відміняйте останній крок). Астри немає під рукою, можу протестувати на OceanWP.

**Comment by ElonaA** [comment]
Astra безкоштошна теж є серед популярних. Потестила, у вас так само перекидує на сторінку /checkout/order-pay/7727/?key=wc_order_ а вже потім до шлюза. Просто у вас це трохи швидше відбувається, та ви не бачите тієї сторінки.

**Comment by iWpress** [comment]
Перевірив OceanWP, той дісно відправляє на сторінку Оплата замовлення:
https://iwp.pp.ua/index.php/checkout-2/order-pay/1142/?key=wc_order_ACy8pnsEzBL31&order=1142
і сам перекидує оплату:
[Останній скріншот](https://imgur.com/a/wVo0XoS)

**Comment by ElonaA** [comment]
В самому плагіні вейфопей у коді стоїть перехід на цю сторінку, де схована кнопка сабміту, яка спрацьовує по сеттаймауту через 200мс. Тому це саме цей плагін. А у підтримці кажуть ми цей плагін не робили, зверніться до розробників вукомерц.

**Comment by iWpress** [comment]
Цікаво, я код майже не дивився.

**Comment by ElonaA** [comment]
Мені в сапорті на цьому тижні сказали що: 
"На жаль ми не розробляли  даний плагін особисто, тому радимо Ваше питання задати саме до розробників, даного сервісу WooCommerce". Та зверніться до розробників плагіну. ТОбто він у них на сайті є, але відношення до нього нібито вони не мають.

**Comment by iWpress** [comment]
Дякую за інформацію. Треба якось зв'язатися з розробниками, щоб вони затвердили мій реквест.

**Comment by ElonaA** [comment]
На жаль не знаю, написала сюда з надією на допомогу, але бачу що тут ніхто не відповідає.

**Comment by iWpress** [comment]
Я написав одному з авторів, він відповів, що давно цим не займається. Порадив написати у группі Facebook. До того я писав у підтримку ел. поштою. Зараз ще написав у їхній Telegram (https://t.me/WFPsupport_bot). Відповіли і написали, що отримали мого листа поштою, пом'ятають, але ще не встигли перевірити та затвердити. Спробуйте.

**Comment by iWpress** [comment]
На жаль, це традеціно для нас, зробити щось гарне, цікаве та закинути. Взагалі то, в мене немає зауважень до плагіну, свою функцію він виконує. Немає підтримки, то погано. Якщо Вам це не дуже срочно, спробую подивитися. Ось мій Гіт, тут можно забрати здагін з моїми правками, доки його не затвердять тут.

**Comment by ElonaA** [comment]
Ота відповідь, що це не наше, знайдіть розробників була якраз з підтримки бота в телеграмі) Дякую, а мені треба звичайно на "завтра" тому пока щось придумаю. Дякую, локалізація знадобиться якраз.

**Comment by lifeact** [comment]
@iWpress   Вітаю! 

Вибачаюсь що втручаюсь...    А Ви не підкажете будь-ласка по моєї ситуації...  я коли тестував "оплата при доставці" у мене відпрацевував такий код:

`

/**
 * Auto Complete all WooCommerce orders.
 */
add_action( 'woocommerce_thankyou', 'custom_woocommerce_auto_complete_order' );
function custom_woocommerce_auto_complete_order( $order_id ) { 
    if ( ! $order_id ) {
        return;
    }

    $order = wc_get_order( $order_id );
    $order->update_status( 'completed' );
}


add_action( 'woocommerce_order_status_completed', 'order_completed', 1);

function order_completed($order_id) {
		
		$order = wc_get_order( $order_id );
		$user_id = $order->get_user_id(); // or $order->get_customer_id();	// can use the _customer_user meta key $user_id = get_post_meta($order_id, '_customer_user', true);	
		$items = $order->get_items();
		
		.......
`

а коли додав цей платіжний модуль, то хук  woocommerce_thankyou не відпрацевує, модіфікував його так: 

`
add_action( 'woocommerce_thankyou', 'custom_woocommerce_auto_complete_order', 20, 1 );
`

не допомогло
